extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRuntimeConfig.java

```diff
@@ -0,0 +1,212 @@
+package io.quarkus.oidc.runtime;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class OidcRuntimeConfig {
+
+    /**
+     * The maximum amount of time the adapter will try connecting to the currently unavailable OIDC server for.
+     * For example, setting it to '20S' will let the adapter keep requesting the connection for up to 20 seconds.
+     */
+    @ConfigItem
+    public Optional<Duration> connectionDelay;
+
+    /**
+     * The base URL of the OpenID Connect (OIDC) server, for example, 'https://host:port/auth'.
+     * All the other OIDC server page and service URLs are derived from this URL.
+     * Note if you work with Keycloak OIDC server, make sure the base URL is in the following format:
+     * 'https://host:port/auth/realms/{realm}' where '{realm}' has to be replaced by the name of the Keycloak realm.
+     */
+    @ConfigItem
+    Optional<String> authServerUrl;
+    /**
+     * Relative path of the RFC7662 introspection service.
+     */
+    @ConfigItem
+    Optional<String> introspectionPath;
+    /**
+     * Relative path of the OIDC service returning a JWK set.
+     */
+    @ConfigItem
+    Optional<String> jwksPath;
+    /**
+     * Public key for the local JWT token verification.
+     */
+    @ConfigItem
+    Optional<String> publicKey;
+    /**
+     * The client-id of the application. Each application has a client-id that is used to identify the application
+     */
+    @ConfigItem
+    Optional<String> clientId;
+    /**
+     * Configuration to find and parse a custom claim containing the roles information.
+     */
+    @ConfigItem
+    OidcRuntimeConfig.Roles roles;
```

Comment:
Can `OidRuntimeConfig.` qualifier be dropped for this and few follow up properties ?

Reply:
Not sure what you mean ... Could you elaborate more please ?

Reply:
@pedroigor I mean, can it just be typed as 
`Roles roles;` ?

Reply:
Ahh, right. I'll change, sorry ...